bearer/bearer.go

```diff
@@ -233,9 +233,18 @@ func (b *Token) SetEACLTable(table eacl.Table) {
 	b.eaclTableSet = true
 }
 
+// GetEACLTable returns extended ACL carried by the Token. Returns false if the
+// eACL is unset.
+func (b Token) GetEACLTable() (eacl.Table, bool) {
+	return b.eaclTable, b.eaclTableSet
+}
+
 // EACLTable returns extended ACL table set by SetEACLTable.
 //
 // Zero Token has zero eacl.Table.
+//
+// Deprecated: zero property of [eacl.Table] is unrecognizable, use
```

Comment:
It's much easier to _always_ have some table. Can we have a token now without it? Not a real one for sure.

Reply:
> Can we have a token now without it?

there is not cases allowed by the protocol. But may be in some stages of the user app: zero instance, after `Unmarshal`, etc.

Reply:
Well, then zero instance should have an empty table.

Reply:
it has, but we cannot determine table emptiness

Reply:
Why do you care about emptiness? What's wrong with the definition of empty table from https://pkg.go.dev/github.com/nspcc-dev/neofs-sdk-go/eacl#NewTable?

Reply:
i dont care about table emptiness, i care about the presence which should be determined somehow. Boolean flag is the one-condition testable way. Other one could be pointer return which i dont like

Reply:
For what purpose? All proper tokens have tables. Token without a table doesn't exists. Can we make it like that?

Reply:
blank instance with unset table

Reply:
> blank instance with unset table

Blank instances should have an empty table. Don't allow to create it and problem solved. SDK interfaces should be easy to use.

Reply:
thats it, there is **no** easy interface to check table for zero (incl. absence). This could be `IsZero` interface which could replace proposed method. https://pkg.go.dev/github.com/nspcc-dev/neofs-sdk-go/eacl#NewTable does not fit cuz it returns particular non-zero table and checking field-by-field isnt be forward compatible

Reply:
I don't understand the problem. Look at how `EACLTable()` is used by the node, for example. All of these cases will do `t, _ = GetEACLTable()`. No code cares about this bool. No code should care.

Reply:
You can drop `eaclTableSet` altogether. No one would notice.

Reply:
> No code should care.

table can be omitted in the just created instance or in the protocol message. This case should be observable. The fact that some code may rely on indirect signs of absence, such as the omission of a signature, does not mean that no one cares whether the table is specified. I'd like to see the case when smbd has completely forgotten the table, although this can be drawn to a more specific "there is something wrong with the token...". Makes easier to understand the protocol violations

> You can drop eaclTableSet altogether. No one would notice.

i know, it's private

Reply:
> just created instance

Should always have it.

> in the protocol message

Decoding error (that we have already anyway).

Reply:
> Should always have it.

there is no way to ensure this

Reply:
what code except unmarshalers needs this information? `ReadFromV2` just requires it

Reply:
API client ops may require parameterized token validation

Reply:
API client always requires v2 version inside, right? my main concern is usability. i feel like there will be no one (or literally a single strange case) `ok` flag usage in practice but also everybody will be forced to suppress this flag

it was always a pain in some part of me to use skip flags like in https://github.com/nspcc-dev/neofs-sdk-go/blob/3a8e90ffe366f0f562878c830de8a700265736cf/object/object.go#L246 and now we have https://github.com/nspcc-dev/neofs-sdk-go/blob/3a8e90ffe366f0f562878c830de8a700265736cf/object/object.go#L267

years of suffering to understand that it was not necessary. nothing changed in fact, we just have admitted that zero ID is not possible _in practice_ and added some words somewhere. the same way i do not see what user code _in practice_ will use it. even if it is required, thousands of `table, _` and a single usecase are worse to me than a separate method added after we have found it is required by something

Reply:
yeah, zero prop is very good. Note that it's stated even [now](https://pkg.go.dev/github.com/nspcc-dev/neofs-sdk-go/bearer#Token.EACLTable). But then the question is - how to check? Zero IDs are easy to detect and we added dedicated [method](https://github.com/nspcc-dev/neofs-sdk-go/blob/3a8e90ffe366f0f562878c830de8a700265736cf/container/id/id.go#L153). Moreover, ID types are comparable and can be zerochecked via `==`. But most of the structs like eACL cannot

it should be noted that zeroness removes the diff b/w a complete absence of a field and a zero but specified one. Such a difference can only be emphasized by an additional flag. Not a eACL case

so, as i said above, im all in for returning struct only, but with exposed `IsZero` method. The code would be:
```go
eACL := token.EACLTable()
if eACL.IsZero() { // only if i care
  // fail
}
// other stuff
```

---

to sum up, i see next options:
1. change nothing. Always seems nicer, but the fact of the table absence will continue to merge with indirect problems of "there is no such or such field"
2. as proposed now (mb with dropped deprecation mark and keeping both methods and letting user to decide what to use)
3. add `eacl.Table.IsZero` method and leave token untouched. Reuse ID approach and, again, let user decide to check or not to check. It will be very leghtweight btw

my choice is 3

Reply:
ok with 3 too. @roman-khimov

Reply:
OK for `IsZero`.

Reply:
done